### ISSUE TITLE
Bug: Forcing 2FA causes users to be unable to set their own 2FA credentials.

### ISSUE THREAD

**Issue by benlk** [Automattic/vip-go-mu-plugins#4475]
## Expected/Desired Behavior

The expected behavior is that when a site is running with `add_filter( 'wpcom_vip_enable_two_factor', '__return_true' );`, they will still be able to enroll themselves in 2FA as a self-service option.

## Actual Behavior

The user receives a permissions error.

![Screenshot 2023-05-18 at 17 31 35](https://github.com/Automattic/vip-go-mu-plugins/assets/1754187/8af27309-9103-4c5e-850d-a8ff99fcdd3d)

## Steps to Reproduce the Problem

1. Add `add_filter( 'wpcom_vip_enable_two_factor', '__return_true' );` to your site
1. Create a new user with administrator, editor, author, or other elevated permissions
2. Log into the site as the user
3. See the "Two Factor Authentication is required to edit content on this site." banner
4. Click the "Enable Two-factor Authentication" link
5. Configure TOTP with your TOTP app
6. Input a test code and hit "Submit"
7. Receive error message.

Another user with `edit_users` permission can submit the code on behalf of the affected user, as a workaround. Screenshariung or desk-sharing works to allow the affected user to view the QR code for TOTP setup.

## (Optional) Additional notes

This is fixed in https://github.com/WordPress/two-factor/pull/560 , which will be in version 0.9 of the two-factor plugin.

VIP doesn't need to do anything to fix this, beyond upgrading to the new version of `two-factor` when it's released: https://github.com/WordPress/two-factor/milestone/24

I'm just submitting this issue so that you're aware and are tracking it.

**Comment by spenserhale** [comment]
Leaving the note that it is fixable by filtering the API Route config:

```php
function hooks_filter_two_factor_endpoint_configuration(array $endpoints): array {
    if ( isset($endpoints['/two-factor/1.0/totp']) ) {
        $endpoints['/two-factor/1.0/totp'][0]['args']['user_id']['type'] = 'integer';
        $endpoints['/two-factor/1.0/totp'][1]['args']['user_id']['type'] = 'integer';
    }

    if ( isset($endpoints['/two-factor/1.0/generate-backup-codes']) ) {
        $endpoints['/two-factor/1.0/totp'][0]['args']['user_id']['type'] = 'integer';
    }

    return $endpoints;
}
add_filter( 'rest_endpoints', 'hooks_filter_two_factor_endpoint_configuration' );
```

I added this filter to our WordPress installs at WP VIP to resolve while we wait for 0.9 to be released.

**Comment by benlk** [comment]
This is still waiting on release 0.8.2 of WordPress/two-factor: https://github.com/WordPress/two-factor/milestone/27?closed=1

**Comment by sjinks** [comment]
I believe this was addressed in #4684; could you please check?

Thank you!

**Comment by benlk** [comment]
Will check this when I have a chance.
